internal/hcl/evaluator.go

```diff
@@ -634,7 +637,11 @@ func (e *Evaluator) loadModule(b *Block) (*ModuleCall, error) {
 	if e.moduleMetadata != nil {
 		// if we have module metadata we can parse all the modules as they'll be cached locally!
 		for _, module := range e.moduleMetadata.Modules {
+			// Strip any "module." and "[*]" parts from the module name so it matches the manifest key format
 			key := modReplace.ReplaceAllString(b.FullName(), "")
+			key = nestedModReplace.ReplaceAllString(key, ".")
+			key = modArrayPartReplace.ReplaceAllString(key, "")
```

Comment:
Could we hoist this above the `range` as we don't need to do this for every loop (I know this was like this before, but it seems best to refactor now)

Then if we could add a `Debug` line on line `649` using the `key` and whatever `modulePath` has evaluated to. This might help tracking down problematic keys in the future.

Reply:
👍

Reply:
Done